### ISSUE TITLE
Emit errors for rogue jsdoc tags

### ISSUE THREAD

**Issue by jespertheend** [microsoft/TypeScript#53104]
# Suggestion

## 🔍 Search Terms

List of keywords you searched for before creating this issue. Write them down here so that others can find this suggestion more easily and help provide feedback.
jsdoc rogue type tag

## ✅ Viability Checklist

My suggestion meets these guidelines:

* [ ] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [x] This wouldn't change the runtime behavior of existing JavaScript code
* [x] This could be implemented without emitting different JS based on the types of the expressions
* [x] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, new syntax sugar for JS, etc.)
* [x] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

## ⭐ Suggestion

Emit an error for jsdoc tags that have an unusual or unexpected position.

## 📃 Motivating Example

Say you want to cast a value but you forget the brackets around the value you want to cast:
```js
const x = /** @type {you can pretty much type anything here} */ 3;
```
At the moment no error is emitted at all, but it seems like this shouldn't be legal.

## 💻 Use Cases

```js
/**
 * @param {any} data
 */
function sanitizeData(data) {
    // ... sanitize
    return /** @type {SanitizedType} */ data;
}
```
It's easy to miss that this function has `any` as return type, rather than the `SanitizedType` you'd expect.

---

Note that even though this would be a breaking change, this change would likely only highlight code that was already broken and not type safe, rather than introduce new issues.

Related #47222
